### PR TITLE
Add delegates and handler for didSelectRowAt and shouldHighlightRowAt

### LINKED ISSUE #153: TableView should default to true for shouldHighlightRowAtIndexPath
Expected
```swift
public func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
    shouldHighlightRowAtIndexPath?(indexPath) ?? true
}
```

Actual
```swift
public func tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool {
    shouldHighlightRowAtIndexPath?(indexPath) ?? false
}
```

Reason 

To allow usage of `didSelectRowAtIndex` by default.
```swift
public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
    didSelectRowAtIndexPath?(indexPath)
}
```

### PR DESCRIPTION
**Pull request by ciccioska**
Added tableview delegates:

- didSelectRowAt

- shouldHighlightRowAt

and related handler

Closes #153

### CODE REVIEW

File: Sources/SwiftUIKit/Views/TableView.swift [hunk 1]
```diff
@@ -32,6 +32,8 @@ public typealias TableViewCell = DataConfigurable & CellConfigurable & CellUpdat
 
 public typealias TableHeaderFooterViewHandler = (Int) -> UIView?
 public typealias TableHeaderFooterTitleHandler = (Int) -> String?
+public typealias TableDidSelectIndexPathHandler = (IndexPath) -> Void
+public typealias TableHighlightIndexPathHandler = (IndexPath) -> Bool?
```
Comment (0xLeif) [inline Sources/SwiftUIKit/Views/TableView.swift:36]: This shouldn't need to be optional. 

`tableView(_ tableView: UITableView, shouldHighlightRowAt indexPath: IndexPath) -> Bool`

**Comment by 0xLeif** [comment]
@ciccioska Thanks for making this change! 

Feel free to join [**oneleif**](https://github.com/oneleif)!
- [Discord](https://discord.com/invite/Ay2A8kN)

**Comment by 0xLeif** [comment]
@ciccioska I switched it from **master** to **develop**... Seems like there are some conflicts now. :(

File: Sources/SwiftUIKit/Views/TableView.swift (hunk 1 above)
Comment (ciccioska) [reply Sources/SwiftUIKit/Views/TableView.swift]: You are right, I solved it!

**Pull request merged**